File: trunk/iOS/SensorLayer/SensorLayer/SensorReading.m
//
//  SensorReading.m
//  SensorLayer
//
//  Created by Arjun Ahuja on 03/08/12.
//  Copyright (c) 2012 arjunahuja@iiitd.com. All rights reserved.
//

#import "SensorReading.h"

@implementation SensorReading{
}
-(id)initWithValues: (int)SensorTypeP : (long)TimeStampP : (NSMutableArray *) Values{
    Readings = Values;
    SensorType = SensorTypeP;
    TimeStamp = TimeStampP;
    
    return self;
}
-(NSMutableArray *)getReadings{
    return Readings;
}
@end

File: trunk/Android/src/sensorLayer/SensorReading.java
package sensorLayer;

import java.util.ArrayList;

public class SensorReading{
	
	protected int SensorType;
	protected long TimeStamp;
	protected ArrayList<Double> Readings;
	
	public SensorReading(int SensorType, long TimeStamp, double ...values)
	{
		this.SensorType = SensorType;
		this.TimeStamp = TimeStamp;
		Readings = new ArrayList<Double>();
		
		for(int i=0; i<values.length; i++)
		{
			Readings.add(new Double(values[i]));
		}
	}
	
	public int getSensorType()
	{
		return SensorType;
	}
	
	public long TimeStamp()
	{
		return TimeStamp;
	}
	
	public ArrayList<Double> getReadings()
	{
		return Readings;
	}
	
	

}
